NXTViSy.nxc

```diff
@@ -11,26 +11,27 @@ void SetSensorNXTViSy(byte port)
     NXTViSy_port = port;
     SetSensorLowspeed(port);
 }
-bool NXTViSyGetData(int &distance, byte &action)
+byte NXTViSyGetData(int &distance, byte &action)
 {
     byte received_bytes_count = 0x03;
-    byte inBuffer[] = { I2C_ADDRESS, GET_DATA };
+    byte inBuffer[2] = { I2C_ADDRESS, GET_DATA };
     byte outBuffer[];
+    long error;
     
     while (I2CCheckStatus(NXTViSy_port) != NO_ERR)
         ;
-    if (I2CBytes(NXTViSy_port, inBuffer, received_bytes_count, outBuffer) != NO_ERR)
-        return false;
+    if (I2CBytes(NXTViSy_port, inBuffer, received_bytes_count, outBuffer) == false)
+        return 0x01;
```

Comment:
@vargac if we want to have different return values that would signalize different errors or success they should be defined as constants. This way it is very hard to interpret what `0x01` actually means.